packages/babel-helper-create-class-features-plugin/src/fields.js

```diff
@@ -100,6 +100,34 @@ const privateNameVisitor = {
   },
 };
 
+const privateInVisitor = {
+  BinaryExpression(path) {
+    const { operator, left, right } = path.node;
+    if (operator !== "in") return;
+    if (!path.get("left").isPrivateName()) return;
+
+    const { loose, privateNamesMap } = this;
+    const { name } = left.id;
+
+    if (loose) {
+      const { id } = privateNamesMap.get(name);
+      path.replaceWith(template.expression.ast`
+        Object.prototype.hasOwnProperty.call(${right}, ${id})
+      `);
+      return;
+    }
+
+    const { id, static: isStatic } = privateNamesMap.get(name);
```

Comment:
The private name is not necessarily defined in current class scope.
```js
class A {
  #p;
  B = class B {
    #q;
    m() { #q in B }
  }
}
```
When traversing into `#q in B` while we are still at `class A`, `privateNamesMap.has("q")` will return `undefined`.

---

It can be even trickier given that private name can scatter in nested class.
```js
class B {
  static #p = 42;
  static C = class C {
    static #p = 42;
    static pIn(obj) {
      return obj.#p;
    }
  }
  static pIn (obj) {
    return B.C.pIn(obj)
  }
}
```
Intuitively both `B.pIn(B)` and `B.pIn(B.C)` should return `true` since both `B` and `B.C` has declared `#p`.

Reply:
In the latter case, `#p in B` would be `false`, since each `#p` is a different one - in other words, inside `class C`, there is *no access available* to B's `#p`, because C's `#p` shadows it.

Reply:
Ooooo, this is an excellent catch. It's actually unhandled in the overall private fields visitor! I'm going to open a specific PR for it, and then I'll rebase that into here.

Reply:
This now depends on https://github.com/babel/babel/pull/11405, which we should merge first.